run.go

```diff
@@ -196,61 +196,140 @@ func runWithOptions(suiteName string, runner runner, opt Options) int {
 			for name := range formatters.AvailableFormatters() {
 				names = append(names, name)
 			}
-			fmt.Fprintln(os.Stderr, fmt.Errorf(
+			err := fmt.Errorf(
 				`unregistered formatter name: "%s", use one of: %s`,
 				opt.Format,
 				strings.Join(names, ", "),
-			))
-			return exitOptionError
+			)
+			return ifmt.MultiFormatter{}, err
 		}
 
 		multiFmt.Add(formatterParts[0], out)
 	}
+	return multiFmt, nil
+}
+
+func runsFromPackage(fp string) string {
+	dir := filepath.Dir(fp)
+
+	gopaths := filepath.SplitList(build.Default.GOPATH)
+	for _, gp := range gopaths {
+		gp = filepath.Join(gp, "src")
+		if strings.Index(dir, gp) == 0 {
+			return strings.TrimLeft(strings.Replace(dir, gp, "", 1), string(filepath.Separator))
+		}
+	}
+	return dir
+}
 
-	if opt.ShowStepDefinitions {
+// TestSuite allows for configuration
+// of the Test Suite Execution
+type TestSuite struct {
+	Name                 string
+	TestSuiteInitializer func(*TestSuiteContext)
+	ScenarioInitializer  func(*ScenarioContext)
+	Options              *Options // TODO mutable value - is this necessary?
```

Comment:
With your refactoring and move the function body to one upper level, I can only think the reason for us to keep using pointer here is to avoid incompatible with older version :thinking: 
Looks like they wanted to separate the concerns, test suite initialization vs runner logic to run that suite, so changed to the options configuration within the function body of runner should not reflected to the test suite.

Reply:
yes I suspect you are right - how tolerant should we be of breaking changes.
I had undone some work to avoid breaks in this pr.

Reply:
I'd say we can tolerant to the point that user's expectations and use cases of `godog` library still remain the same. Except for the hook ordering bug we're going to fix, I really don't see we're going to "break" anything else.